source/common/filesystem/filesystem_impl.cc

```diff
@@ -171,12 +172,12 @@ void FileImpl::flushThreadFunc() {
     std::unique_lock<Thread::BasicLockable> flush_lock;
 
     {
-      std::unique_lock<std::mutex> write_lock(write_lock_);
+      Thread::LockGuard write_lock(write_lock_);
 
       // flush_event_ can be woken up either by large enough flush_buffer or by timer.
       // In case it was timer, flush_buffer_ can be empty.
       while (flush_buffer_.length() == 0 && !flush_thread_exit_) {
-        flush_event_.wait(write_lock);
+        flush_event_.wait(write_lock_);
```

Comment:
RFC: I can see a potential objection to this change, because if CondVar::Wait() temporarily releases the lock and then throws, then Thread::LockGuard will call write_lock_.unlock() again.

However I don't believe the underlying absl::CondVar::Wait() can or ever will throw, so I think this is safe. 

Moreover, absl::CondVar::Wait takes a concrete absl::Mutex* and there's no absl equivalent to std::condition_variable_any.

It occurs to me now that I could leave flush_event_ as a std::conditional_variable_any, but I have a preference for staying in the absl ecosystem here. But if reviewer(s) feel strongly I could probably revert the types of the condition variables here.

Reply:
I think it's fine, I doubt that Abseil libraries throw, given Google C++ aversion to exceptions. It would be nice to put this reasoning on the contract of `CondVar` to assist the hapless person debugging some concurrency issue in the future.

Also, maybe add a link to somewhere that demonstrates that Abseil does't throw exceptions; presumably this kind of detail should be provided.

Reply:
added some comments here and in common/common/thread.h in the CondVar declaration.